packages.js

```diff
@@ -0,0 +1,114 @@
+const debug = require('diagnostics')('warehouse:packages');
+
+const defaultOptions = {
+  cache: {}
+};
+
+class Packages {
+  /**
+   * @constructor
+   * @param {Warehouse} warehouse Reference to the Warehouse instance.
+   * @param {Object} [options] Packages options object
+   * @param {Object} [options.cache] An options object for caching package information
+   * @param {boolean} [options.cache.enabled] True if you want to use a cache of package information,
+   * False if you want to hit the service every time
+   * @param {number} [options.cache.interval] An interval in ms on which the cache should be refreshed. (Defaults to 5 minutes)
+   * @public
+   */
+  constructor(warehouse, options) {
+    this.warehouse = warehouse;
+
+    options = Object.assign({}, defaultOptions, options);
+    this.cache = new Map();
```

Comment:
How weird is it going to be for consumers to have different cache retention/refresh logic between the different data?  You could do something with the existing `out-of-band-cache` where you see if you have all the data cached, and if so, you just grab the single package data from that, otherwise you pull the individual package.  I suspect though that we'd need to build some features into that for being able to do simple sets on data.  But I suspect that it'd be worth doing to have more consistency here.

Maybe we save off both the full set and the single packages and update both on refresh of either.